kaldb/src/main/java/com/slack/kaldb/metadata/schema/ChunkSchema.java

```diff
@@ -0,0 +1,74 @@
+package com.slack.kaldb.metadata.schema;
+
+import com.google.common.base.Objects;
+import com.slack.kaldb.metadata.core.KaldbMetadata;
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * This schema class enforces schema for a chunk. The schema is only written in indexer and on the
+ * cache node the schema is read only.
+ *
+ * <p>The fieldMap is a map to the field name and LuceneFieldDef that stores the field definitions.
+ * Since this field is written and read my multiple threads it is hardcoded as a ConcurrentHashMap.
+ * New fields are added to the fieldMap when it's written and read during query.
+ */
+public class ChunkSchema extends KaldbMetadata {
```

Comment:
We seem to be missing the corresponding `ChunkSchemaStore` for persisting this metadata, and the associated test.